src/lib.rs

```diff
@@ -95,3 +97,9 @@ pub trait GenFn4<T>: Fn(&Seed, &Point4<T>) -> T {}
 impl<T, F> GenFn2<T> for F where F: Fn(&Seed, &Point2<T>) -> T {}
 impl<T, F> GenFn3<T> for F where F: Fn(&Seed, &Point3<T>) -> T {}
 impl<T, F> GenFn4<T> for F where F: Fn(&Seed, &Point4<T>) -> T {}
+
+pub trait Module<T> {
```

Comment:
I wonder if `Source` or `ValueSource` would be a better term? `Module` seems rather generic a term. @Aatch?